File: backend/app/schemas/base_task.py
from typing import Optional, List

from pydantic import BaseModel

from app.schemas.slide import Slide
from app.schemas.task import Task


class BaseTaskBase(BaseModel):
    name: Optional[str] = None
    short_name: Optional[str] = None
    enabled: Optional[bool] = False


class BaseTaskCreate(BaseTaskBase):
    name: str
    slide_id: int
    course_id: int
    task_group_id: Optional[int] = None


class BaseTaskUpdate(BaseTaskBase):
    base_task_id: int
    name: Optional[str] = None
    slide_id: Optional[int] = None
    task_group_id: Optional[int] = None


class BaseTaskInDBBase(BaseTaskBase):
    id: Optional[int] = None
    slide_id: Optional[int] = None
    task_group_id: Optional[int] = None

    class Config:
        orm_mode = True


class BaseTaskInDB(BaseTaskInDBBase):
    pass


class BaseTask(BaseTaskInDB):
    slide: Optional[Slide] = None
    percentage_solved: Optional[float]
    task_count: Optional[int]
    new_tasks: Optional[int] = 0
    correct_tasks: Optional[int] = None
    wrong_tasks: Optional[int] = None


class BaseTaskDetail(BaseTask):
    task_group_id: Optional[int] = None
    task_group_short_name: Optional[str] = None
    tasks: Optional[List[Task]] = None

File: frontend/src/model/baseTask.ts
import { Task } from './task';

export interface BaseTask {
  id: number;
  name: string;
  short_name: string;
  course_id?: number;
  slide_id?: string;
  task_group_id?: number;
  task_group_short_name: string;
  task_count?: number;
  enabled: boolean;
  new_tasks?: number;
  tasks: Task[];
  percentage_solved: number;
  correct_tasks?: number;
  wrong_tasks?: number;
}

export interface CreateBaseTask {
  name: string;
  slide_id: string;
  course_id: number;
  task_group_id?: number;
}

export interface UpdateBaseTask {
  base_task_id: number;
  name?: string;
  slide_id?: number;
  course_id?: number;
  task_group_id?: number;
  enabled?: boolean;
}
